webapp/app/js/models/job.js

```diff
@@ -55,19 +57,30 @@ treeherder.factory('ThJobModel', [
                 if(_.has(response.data, 'job_property_names')){
                     // the results came as list of fields
                     //we need to convert them to objects
-                    item_list = _.map(response.data.results, function(elem){
-                        var job_obj = _.object(response.data.job_property_names, elem);
-                        return new ThJobModel(job_obj);
-                    });
+                    item_list = [];
+                    var props = response.data.job_property_names;
+                    var propsLen = props.length;
+                    for (var i = 0, resultsLen = response.data.results.length; i < propsLen; i++) {
```

Comment:
The condition to break the loop should be i < resultsLen 
I think this is the reason why the tests are failing. You probably don't see the problem if you look at `mozilla-central`, since the number of jobs per result set is > 36. On `try` you will likely see errors in the console.